### PR TITLE
Fix mcv husks losing their faction after being captured

### LINKED ISSUE #11491: After repaired from Husk MCV loses its old prerequisites.
As i said in tile. I was Allies and, enemy was Soviet. I killed his MCV repaired it with mechanic but it still provides Allied Structures for me.

I also tested hijacking them for comparasion it works fine. I still get MCVs old owner's building after hijacking.

### PR DESCRIPTION
**Pull request by abc013**
Fixes #11491.

### CODE REVIEW

**Comment by pchote** [comment]
This just pushes the bug down a level: husks of mcvs that were captured from another player will still lose their faction.  You should store the original faction in the constructor and then use that instead.

**Comment by pchote** [comment]
See https://github.com/OpenRA/OpenRA/blob/master/OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs#L49 for an example of how to get the faction.

**Comment by abc013** [comment]
Updated.

**Comment by obrakmann** [comment]
Fix confirmed, :+1:

File: OpenRA.Mods.Common/Traits/TransformOnCapture.cs [hunk 1]
```diff
@@ -20,19 +20,24 @@ public class TransformOnCaptureInfo : ITraitInfo
 		public readonly int ForceHealthPercentage = 0;
 		public readonly bool SkipMakeAnims = true;
 
-		public virtual object Create(ActorInitializer init) { return new TransformOnCapture(this); }
+		public virtual object Create(ActorInitializer init) { return new TransformOnCapture(this, init); }
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/TransformOnCapture.cs]: Should be `init, this`

**Comment by abc013** [comment]
@Phrohdoh changed.

File: OpenRA.Mods.Common/Traits/TransformOnCapture.cs [hunk 2]
```diff
@@ -20,19 +20,24 @@ public class TransformOnCaptureInfo : ITraitInfo
...
+		public virtual object Create(ActorInitializer init) { return new TransformOnCapture(init, this); }
 	}
 
 	public class TransformOnCapture : INotifyCapture
 	{
 		readonly TransformOnCaptureInfo info;
+		readonly string faction;
 
-		public TransformOnCapture(TransformOnCaptureInfo info) { this.info = info; }
+		public TransformOnCapture(TransformOnCaptureInfo info, ActorInitializer init)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/TransformOnCapture.cs]: You also have to change order here then.

**Comment by abc013** [comment]
Thanks!

**Comment by pchote** [comment]
:+1:

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ce0e7ddd03dfe066e80f5bea279e4423f227ffcb

**Pull request merged**